### PR TITLE
Make LevelDemand min_level and max_level optional

### PR DESCRIPTION
**Pull request by visr**
As proposed in https://github.com/Deltares/Ribasim/issues/1428#issuecomment-2076766380

This makes it easier to use LevelDemand in case you only care about a shortage or surplus situation.

### CODE REVIEW

File: core/src/read.jl [hunk 1]
```diff
@@ -93,7 +93,7 @@ function parse_static_and_time(
 
     errors = false
     t_end = seconds_since(config.endtime, config.starttime)
-    trivial_timespan = [nextfloat(-Inf), prevfloat(Inf)]
+    trivial_timespan = [0.0, prevfloat(Inf)]
```
Comment (visr) [inline core/src/read.jl:96]: Without this change we would get a NaN instead of Inf from DataInterpolations.

File: core/src/util.jl
Comment (visr) [inline core/src/util.jl:1]: Returning a storage of 0 for levels below the bottom seems fine to me, and is needed to support -Inf min_level.

File: docs/core/usage.qmd [hunk 2]
```diff
@@ -479,14 +479,17 @@ min_level     | Float64  | $m$          | -
 
 A `LevelDemand` node associates a minimum and a maximum level with connected basins to be used by the allocation algorithm.
 Below the minimum level the basin has a demand of the supplied priority,
-above the maximum level the basin acts as a source, used by all nodes with demands in order of priority.
-The same `LevelDemand` node can be used for basins in different subnetworks.
+above the maximum level the basin has a surplus and acts as a source, used by all nodes with demands in order of priority.
+The same `LevelDemand` node can be used for Basins in different subnetworks.
+
+Both `min_level` and `max_level` are optional, to be able to handle only the demand or surplus side.
+If both are missing `LevelDemand` won't have any effects on allocation.
```
Comment (SouthEndMusic) [inline docs/core/usage.qmd:486]:
```suggestion
If both are missing, `LevelDemand` won't have any effects on allocation.
```

**Review by SouthEndMusic — APPROVED** [review]
Good good

**Pull request merged**